src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -84,32 +84,46 @@ public BucketMetadata getBucketMetadata(String bucketName) {
         return new BucketMetadata(bucketEntity);
     }
 
-    public List<BucketMetadata> listBuckets(List<String> owners, String kind) {
+    public List<BucketMetadata> listBuckets(List<String> owners, String kind, String contentType) {
         if (owners == null) {
             return Collections.emptyList();
         }
 
         List<BucketEntity> entities;
-        if (!StringUtils.isEmpty(kind)) {
+        if (!StringUtils.isEmpty(kind) && !StringUtils.isEmpty(contentType)) {
+            entities = bucketRepository.findByOwnerIsInContainingKindAndContentType(owners, kind, contentType);
+        } else if (!StringUtils.isEmpty(kind) && StringUtils.isEmpty(contentType)) {
             entities = bucketRepository.findByOwnerIsInContainingKind(owners, kind);
-        } else {
+        } else if (StringUtils.isEmpty(kind) && !StringUtils.isEmpty(contentType)) {
+            entities = bucketRepository.findByOwnerIsInContainingKind(owners, kind);
+        }
+
+        else {
             entities = bucketRepository.findByOwnerIn(owners);
         }
 
         log.info("Buckets size {}", entities.size());
         return entities.stream().map(BucketMetadata::new).collect(Collectors.toList());
     }
 
-    public List<BucketMetadata> listBuckets(String ownerName, String kind) {
+    public List<BucketMetadata> listBuckets(String ownerName, String kind, String contentType) {
```

Comment:
The same comment as before for method
listBuckets(List<String> owners, String kind, String contentType

if we make ownerName, kind, contentType -  Optional, then it will be easier and more clear to check paramters